### ISSUE TITLE
Pong the Next Level (GBC)

### ISSUE THREAD

**Issue by Keltron3030** [meleu/RetroArch-problematic-cheevos#36]
Game: 
http://retroachievements.org/game/5067

Achievement:
http://retroachievements.org/achievement/69294

and all achievements involving the following addresses.

`0x00dff4: Difficulty [RAlibretro]`
`0x00bf84: Modes [RAlibretro]`
`0x00fff4: Difficulty [00=easy][01=medium][02=hard] [VBA]`
`0x00df84: Modes [VBA]`

Alts were created for the mode and difficulty addresses to allow for compatibility in RAVBA and Ralibretro; however there appears to still be an address compatibility issue using certain Retroarch 1.7.5 builds with Gambatte v0.5.0-netlink.

This bug affects RetroArch 1.7.5(have not tried earlier versions), specifically Android and PC builds running the Gambatte core.  The Raspi build works fine, and the build for switch work fine according to Mr. Ownership. 

Achievements using just these addresses work fine with the same problem setups listed above.

`0x00c24c: p1 points`
`0x00c24f: [01=you won] [02=try again]`

So to summarize, the issue is not specifically a problem with RetroArch 1.7.5 Android since that build works ok with GearBoy core, nor is it an issue with Gambatte v0.5.0-netlink because that core works fine with RetroArch 1.7.5 Raspi build.

The issue is isolated to only certain builds of Retroarch 1.7.5 (Android and PC) running the Gambatte v.0.5.0-netplay core and in those cases only some addresses display the compatibility problem.

The following issue may help to define the address mapping problem:

https://github.com/libretro/gambatte-libretro/issues/122

**Comment by MrOwnership** [comment]
I am seeing the same issue for the following game:
[http://retroachievements.org/game/13829](url)

Using the following achievement as an example:
[http://retroachievements.org/achievement/70660](url)

It checks if the value at address `0x00defe` is equal to 6, indicating that the user has found the sound test screen. This achievement will trigger on RAVBA and Gambatte v0.5.0 on my Nintendo Switch (RetroArch 1.7.5). The achievement does not trigger on Gambatte v0.5.0-netlink on my PC (RetroArch 1.7.5). I tried finding the address using the built in cheat engine, and then creating an alt for that address (`0x001efe`) but the achievement still would not trigger on Gambatte v0.5.0-netlink on my PC.

This issue is seen on multiple achievements for this game.

**Comment by rzumer** [comment]
I'd like someone to provide a save state immediately prior to what should result in an unlock, so that it is possible to debug and test easily.

**Comment by Keltron3030** [comment]
https://drive.google.com/folderview?id=1--vsKvsy0SJnvNFKA0DRTAJMH5N9fLzD

This is for the achievement mentioned above saved with Gambatte I was running on Android.

**Comment by rzumer** [comment]
Fixed by https://github.com/libretro/gambatte-libretro/pull/124. Expect delays before it is reflected in the distributed cores.

**Comment by meleu** [comment]
closing this as fixed. @keltron3030 feel free to reopen if the problem still happens with you **when using uptodate versions of the mentioned cores**.

and thanks @rzumer for all those fixes!
